src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php

```diff
@@ -23,15 +24,17 @@ class FirewallContext
 {
     private $listeners;
     private $exceptionListener;
+    private $logoutListener;
 
-    public function __construct(array $listeners, ExceptionListener $exceptionListener = null)
+    public function __construct(array $listeners, ExceptionListener $exceptionListener = null, LogoutListener $logoutListener = null)
```

Comment:
fun fact: on 3.4 & up, this class [already has a 3rd argument](https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/SecurityBundle/Security/FirewallContext.php#L28): `FirewallConfig $config`
To prevent any funnier things, I suggest adding this argument in 2.7 (but ignoring its value).
WDYT? Any better idea?

Reply:
Can we do this on concerned branches?

Reply:
we could change for a setter on 2.7, not sure it's better. Here is what you propose #27280

Reply:
I finally managed to merge, see https://github.com/symfony/symfony/commit/86a9c73026111619a89f5cda5c7095bfb7a0f62e#diff-cf0390616c24425d612bcf9ee1555111
@chalasr when doing your PR against master, please also deprecate passing a FirewallConfig as 3rd arg there.